### PR TITLE
Rework of documentation in problem classes

### PR DESCRIPTION
**Pull request by lisawim**
I reworked the problem classes (#286). I wrote some more detailed documentation for some problem classes. For all classes I also reworked the documentation for the class methods. Unfortunately, there is still much to do, but I think we are much closer to finish the issue #286. 

I have no knowledge about many of the problems.. @tlunet told me documentation for problem classes can be reworked if someone will work with that but I guess some problem classes will be untouched in the near future. I could do the rest but I need references for the problems to finish up the documentations to write down the equations for them..

### CODE REVIEW

File: pySDC/implementations/problem_classes/AdvectionDiffusionEquation_1D_FFT.py [hunk 1]
```diff
@@ -7,50 +7,60 @@
...
-            if key not in problem_params:
-                msg = 'need %s to instantiate problem, only got %s' % (key, str(problem_params.keys()))
-                raise ParameterError(msg)
+        # invoke super init, passing number of dofs, dtype_u and dtype_f
+        super().__init__(init=(nvars, None, np.dtype('float64')))
+        self._makeAttributeAndRegister('nvars', 'c', 'freq', 'nu', 'L', localVars=locals(), readOnly=True)
 
         # we assert that nvars looks very particular here.. this will be necessary for coarsening in space later on
-        if (problem_params['nvars']) % 2 != 0:
+        if (self.params.nvars) % 2 != 0:
```
Comment (tlunet) [inline pySDC/implementations/problem_classes/AdvectionDiffusionEquation_1D_FFT.py:61]: Are you sure this works ? `params` is supposed to be a dictionnary only now ...

Comment (tlunet) [reply pySDC/implementations/problem_classes/AdvectionDiffusionEquation_1D_FFT.py:61]: PS : what was said during the pySDC workshop was actually wrong :sweat_smile:

**Comment by tlunet** [comment]
Nice work !

**Comment by tlunet** [comment]
Seems like the pySDC/implementations/problem_classes/AdvectionDiffusionEquation_1D_FFT.py was not tested ... I've added this to the list of problem not tested yet in #223

**Comment by pancetta** [comment]
Great work, thanks a lot for this!

**Pull request merged**